wandb/sdk/wandb_artifacts.py

```diff
@@ -529,6 +529,25 @@ def do_write(f: IO) -> None:
 
         return entry
 
+    def items(self) -> Iterable[Tuple[str, ArtifactEntry]]:
+        if self._logged_artifact:
```

Comment:
Is this triggered just when I am manipulating an artifact after it has been logged but before the run has ended, or also when I manipulate an artifact after the run is over?

Reply:
This is triggered on any manipulation after the artifact has been logged using `log_artifact`. After the artifact is logged, it becomes immutable so we only want to allow read-only access after that point.